app/controllers/galleries_controller.rb

```diff
@@ -40,7 +42,25 @@ def update
   private
   
   def gallery_params
-    params.require(:gallery).permit(:name, :description)
+    params.require(:gallery).permit(:name, :description, :private)
+  end
+  
+  def correct_user
+    @gallery = Gallery.find(params[:id])
+    if @gallery.user != current_user
+      redirect_to root_url
+    end
+  end
+  
+  def private_gallery
+    @gallery = Gallery.find(params[:id])
+    if @gallery.private?
+      if @gallery.user == current_user
+        @gallery
+      else
+        redirect_to root_url
+      end
+    end
```

Comment:
The logic in a before filter should be either `redirect_to` or nothing. In the case that this is a public gallery, just let the request continue. Similarly, if the user owns the gallery, then just let the request continue. Also, same as with `correct_user`, you don't want to use instance variables with before filters. All together, I would write this as:

``` ruby
def user_can_view_gallery
  gallery = Gallery.find(params[:id])
  if gallery.private? && gallery.user != current_user
    redirect_to root_url
  end
end
```